pkg/frontend/adminactions/resources_list_test.go

```diff
@@ -24,111 +24,129 @@ import (
-				resources.EXPECT().ListByResourceGroup(gomock.Any(), "test-cluster", "", "", nil).Return([]mgmtfeatures.GenericResourceExpanded{
-					{
-						Name: to.StringPtr("vm-1"),
-						ID:   to.StringPtr("/subscriptions/id"),
-						Type: to.StringPtr("Microsoft.Compute/virtualMachines"),
-					},
-					{
-						Name: to.StringPtr("storage"),
-						ID:   to.StringPtr("/subscriptions/id"),
-						Type: to.StringPtr("Microsoft.Storage/storageAccounts"),
-					},
-				}, nil)
+			mocks: func(tt *test, resources *mock_features.MockResourcesClient, virtualMachines *mock_compute.MockVirtualMachinesClient, virtualNetworks *mock_network.MockVirtualNetworksClient, routeTables *mock_network.MockRouteTablesClient, diskEncryptionSets *mock_compute.MockDiskEncryptionSetsClient) {
+				defaultListByResourceGroup(resources)
+				defaultVirtualMachines(virtualMachines)
 
+				// Fail virtualNetworks with a GET error
 				virtualNetworks.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(mgmtnetwork.VirtualNetwork{}, fmt.Errorf("Any error during Get, expecting a permissions error"))
-
-				resources.EXPECT().GetByID(gomock.Any(), "/subscriptions/id", azureclient.APIVersion("Microsoft.Storage")).Return(mgmtfeatures.GenericResource{
-					Name:     to.StringPtr("storage"),
-					ID:       to.StringPtr("/subscriptions/id"),
-					Type:     to.StringPtr("Microsoft.Storage/storageAccounts"),
-					Location: to.StringPtr("eastus"),
-				}, nil)
-
-				virtualMachines.EXPECT().Get(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(mgmtcompute.VirtualMachine{
-					ID:   to.StringPtr("/subscriptions/id"),
-					Type: to.StringPtr("Microsoft.Compute/virtualMachines"),
-					VirtualMachineProperties: &mgmtcompute.VirtualMachineProperties{
-						ProvisioningState: to.StringPtr("Succeeded"),
-					},
-				}, nil)
 			},
 			wantResponse: []byte(`[{"properties":{"provisioningState":"Succeeded"},"id":"/subscriptions/id","type":"Microsoft.Compute/virtualMachines"},{"id":"/subscriptions/id","name":"storage","type":"Microsoft.Storage/storageAccounts","location":"eastus"}]`),
 		},
+		{
+			name: "enabled diskencryptionsets",
+			mocks: func(tt *test, resources *mock_features.MockResourcesClient, virtualMachines *mock_compute.MockVirtualMachinesClient, virtualNetworks *mock_network.MockVirtualNetworksClient, routeTables *mock_network.MockRouteTablesClient, diskEncryptionSets *mock_compute.MockDiskEncryptionSetsClient) {
+				defaultListByResourceGroup(resources)
+				defaultVirtualNetworks(virtualNetworks, routeTables, mockSubID)
+				defaultVirtualMachines(virtualMachines)
+				defaultDiskEncryptionSets(diskEncryptionSets)
+			},
+			diskEncryptionSetId: fmt.Sprintf("/subscriptions/%s/resourceGroups/test-cluster/providers/Microsoft.Network/virtualNetworks/test-vnet/subnets/master", mockSubID),
```

Comment:
So is it disk encryption set Id or vnet id? And how does it correspond to what we expect in `diskEncryptionSets.EXPECT().Get(...)` and in `wantResponse`?

Same in the next test case.

Reply:
Updated id, that information is not used in the function, only the resourceGroup and resourceName, which wasn't being trapped because of `gomock.Any()`